run/test/specs/utils/check_colour.ts

```diff
@@ -23,3 +24,28 @@ export async function parseDataImage(base64: string) {
   // console.info("Middle x:", middleX, "middleY:", middleY, "width:", width);
   return pixelColor;
 }
+
+// Function to compare two colors within a specified CIEDE2000 tolerance
+export function compareColors(
+  hex1: string,
+  hex2: string,
+  tolerance: number = 2.3 // looks-same default value which is "enough for most cases"
+): boolean {
```

Comment:
no need to type this, the type is inferred correctly from the return, and this just becomes one more type to maintain, long term